### PR TITLE
*: update syncer document

### PR DESCRIPTION
**Pull request by GregoryIan**
@lilin90 @buggithubs PTAL

### CODE REVIEW

File: op-guide/migration.md [hunk 1]
```diff
@@ -153,28 +153,51 @@ binlog-gtid = "2bfabd22-fff7-11e6-97f7-f02fa73bcb01:1-23,61ccbb5d-c82d-11e6-ac2e
...
 
 ## meta 文件地址
 meta = "./syncer.meta"
-
 worker-count = 16
-batch = 10
+batch = 1000
+flavor = "mysql"
 
 ## pprof 调试地址, Prometheus 也可以通过该地址拉取 syncer metrics
```
Comment (lilin90) [inline op-guide/migration.md]:
```suggestion
## pprof 调试地址，Prometheus 也可以通过该地址拉取 Syncer metrics
```

File: op-guide/migration.md [hunk 2]
```diff
@@ -153,28 +153,51 @@ binlog-gtid = "2bfabd22-fff7-11e6-97f7-f02fa73bcb01:1-23,61ccbb5d-c82d-11e6-ac2e
...
-## 将 127.0.0.1 修改为相应主机 IP 地址
-status-addr = "127.0.0.1:10086"
+status-addr = ":8271"
+
+## 如果设置为 true，Syncer 遇到 ddl 的时候就会停止退出
```
Comment (lilin90) [inline op-guide/migration.md]:
```suggestion
## 如果设置为 true，Syncer 遇到 DDL 语句时就会停止退出
```

File: op-guide/migration.md [hunk 3]
```diff
@@ -153,28 +153,51 @@ binlog-gtid = "2bfabd22-fff7-11e6-97f7-f02fa73bcb01:1-23,61ccbb5d-c82d-11e6-ac2e
...
+stop-on-ddl = false
 
-## 跳过 DDL 或者其他语句，格式为 **前缀完全匹配**，如: `DROP TABLE ABC`,则至少需要填入`DROP TABLE`.
-# skip-sqls = ["ALTER USER", "CREATE USER"]
+## 跳过 DDLs，格式为 **前缀完全匹配**，如: `DROP TABLE ABC`, 则至少需要填入`DROP TABLE`.
```
Comment (lilin90) [inline op-guide/migration.md]:
```suggestion
## 跳过 DDL 语句，格式为 **前缀完全匹配**，如：`DROP TABLE ABC` 至少需要填入`DROP TABLE`
```

File: op-guide/migration.md [hunk 4]
```diff
@@ -153,28 +153,51 @@ binlog-gtid = "2bfabd22-fff7-11e6-97f7-f02fa73bcb01:1-23,61ccbb5d-c82d-11e6-ac2e
...
 ## 在使用 route-rules 功能后，
 ## replicate-do-db & replicate-ignore-db 匹配合表之后(target-schema & target-table )数值
 ## 优先级关系: replicate-do-db --> replicate-do-table --> replicate-ignore-db --> replicate-ignore-table
 ## 指定要同步数据库名；支持正则匹配，表达式语句必须以 `~` 开始
 #replicate-do-db = ["~^b.*","s1"]
 
+## 指定**忽略**同步数据库；支持正则匹配，表达式语句必须以 `~` 开始
+#replicate-ignore-db = ["~^b.*","s1"]
+
+# skip-dmls 支持跳过 DML binlog events. type 字段的值：'insert', 'update' and 'delete'.
```
Comment (lilin90) [inline op-guide/migration.md]:
```suggestion
# skip-dmls 支持跳过 DML binlog events，type 字段的值可为：'insert'，'update' 和 'delete'
```

File: tools/syncer.md [hunk 5]
```diff
@@ -71,15 +71,23 @@ Usage of syncer:
   -c int
       syncer 处理 batch 线程数 (默认 16)
   -config string
-      指定相应配置文件启动 sycner 服务；如 `--config config.toml` 
+      指定相应配置文件启动 sycner 服务；如 `--config config.toml`
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
      指定相应配置文件启动 Sycner 服务；如 `--config config.toml`
```

File: tools/syncer.md [hunk 6]
```diff
@@ -71,15 +71,23 @@ Usage of syncer:
...
+  -enable-ansi-quotes
+      使用 ANSI_QUOTES sql_mode 来解析 sql
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
      使用 ANSI_QUOTES sql_mode 来解析 SQL 语句
```

File: tools/syncer.md [hunk 7]
```diff
@@ -71,15 +71,23 @@ Usage of syncer:
...
+  -flavor string
+      上游数据库实例类型，目前支持 "mysql" 和 "mariadb"
   -log-file string
       指定日志文件目录；如 `--log-file ./syncer.log`
   -log-rotate string
       指定日志切割周期, hour/day (默认 "day")
   -meta string
       指定 syncer 上游 meta 信息文件  (默认与配置文件相同目录下 "syncer.meta")
+  -persistent-dir string
+      指定同步过程中历史 schema 结构的保存文件地址，如果设置为空，则不保存历史 schema 结构；如果不为空，则根据 binlog 里面包含的数据的 column 长度选择 schema 来还原 dml
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
      指定同步过程中历史 schema 结构的保存文件地址，如果设置为空，则不保存历史 schema 结构；如果不为空，则根据 binlog 里面包含的数据的 column 长度选择 schema 来还原 DML 语句
```

File: tools/syncer.md [hunk 8]
```diff
@@ -90,49 +98,57 @@ Syncer 的配置文件 `config.toml`：
...
-batch = 10
+batch = 1000
+flavor = "mysql"
 
 ## pprof 调试地址, Prometheus 也可以通过该地址拉取 syncer metrics
-## 将 127.0.0.1 修改为相应主机 IP 地址
-status-addr = "127.0.0.1:10086"
+status-addr = ":8271"
+
+## 如果设置为 true，Syncer 遇到 ddl 的时候就会停止退出
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
## 如果设置为 true，Syncer 遇到 DDL 语句时就会停止退出
```

File: tools/syncer.md [hunk 9]
```diff
@@ -90,49 +98,57 @@ Syncer 的配置文件 `config.toml`：
...
+stop-on-ddl = false
+
+## 跳过 DDLs，格式为 **前缀完全匹配**，如: `DROP TABLE ABC`, 则至少需要填入`DROP TABLE`.
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
## 跳过 DDL 语句，格式为 **前缀完全匹配**，如：`DROP TABLE ABC` 至少需要填入`DROP TABLE`.
```

File: tools/syncer.md [hunk 10]
```diff
@@ -90,49 +98,57 @@ Syncer 的配置文件 `config.toml`：
...
+# skip-ddls = ["ALTER USER", "CREATE USER"]
 
-# 注意: skip-sqls 已经废弃, 请使用 skip-ddls.
-# skip-ddls 可以跳过与 TiDB 不兼容的 DDL 语句，支持正则语法。
-# skip-ddls = ["^CREATE\\s+USER"]
+## 在使用 route-rules 功能后，
+## replicate-do-db & replicate-ignore-db 匹配合表之后(target-schema & target-table )数值
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
## replicate-do-db & replicate-ignore-db 匹配合表之后 (target-schema & target-table) 的数值
```

File: tools/syncer.md [hunk 11]
```diff
@@ -90,49 +98,57 @@ Syncer 的配置文件 `config.toml`：
...
+## 指定要同步数据库名；支持正则匹配，表达式语句必须以 `~` 开始
+#replicate-do-db = ["~^b.*","s1"]
 
-# 注意: skip-events 已经废弃, 请使用 skip-dmls 
-# skip-dmls 用于跳过 DML 语句. type 字段取值为 'insert', 'update', 'delete'。
-# 下面的例子为跳过 foo.bar 表的所有 delete 语句。
+## 指定**忽略**同步数据库；支持正则匹配，表达式语句必须以 `~` 开始
+#replicate-ignore-db = ["~^b.*","s1"]
+
+# skip-dmls 支持跳过 DML binlog events. type 字段的值：'insert', 'update' and 'delete'.
```
Comment (lilin90) [inline tools/syncer.md]: ditto

File: tools/syncer.md [hunk 12]
```diff
@@ -480,35 +493,51 @@ Syncer 对外提供 metric 接口，需要 Prometheus 主动获取数据。配
...
+- metrics: `rate(syncer_binlog_event_count[1m])`
+- info: 统计 Syncer 每秒已经收到的 binlog 个数
 
-#### title: syncer_binlog_file
+#### title: binlog event transform
 
-- metrics: `syncer_binlog_file`
-- info: Syncer 同步 master binlog 的文件数量
+- metrics: `histogram_quantile(0.8, sum(rate(syncer_binlog_event_bucket[1m])) by (le))`
+- info: Syncer 把 binlog 转换为 SQLs 的耗时
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
- info: Syncer 把 binlog 转换为 SQL 语句的耗时 
```

File: tools/syncer.md [hunk 13]
```diff
@@ -480,35 +493,51 @@ Syncer 对外提供 metric 接口，需要 Prometheus 主动获取数据。配
...
 
-- metrics: `irate(syncer_binlog_skipped_events_total[1m])`
-- info: Syncer 同步 master binlog 文件时跳过执行 SQL 的数量统计。跳过 SQL 语句格式由 `syncer.toml` 文件中的 `skip-sqls` 参数控制。
+- metrics: `rate(syncer_binlog_skipped_events_total[1m])`
+- info: Syncer 跳过的 binlog 的个数，你可以在配置文件中配置 `skip-ddls` 和 `skip-dmls` 来跳过指定的 binlog
+
+#### title: position of binlog position
+
+- metrics: `syncer_binlog_pos{node="syncer"}` and `syncer_binlog_pos{node="master"}`
+- info: 需要配合 `file number of binlog position` 一起看. `syncer_binlog_pos{node="master"}` 表示上游 MySQL 当前 binlog position 的 position 值, `syncer_binlog_pos{node="syncer"}` 表示上游 Syncer 已经同步到的 binlog position 的 position 值
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
- info: 需配合 `file number of binlog position` 一起看。`syncer_binlog_pos{node="master"}` 表示上游 MySQL 当前 binlog position 的 position 值，`syncer_binlog_pos{node="syncer"}` 表示上游 Syncer 已经同步到的 binlog position 的 position 值
```

File: tools/syncer.md [hunk 14]
```diff
@@ -480,35 +493,51 @@ Syncer 对外提供 metric 接口，需要 Prometheus 主动获取数据。配
...
+
+#### title: file number of binlog position
+
+- metrics: `syncer_binlog_file{node="syncer"}` and `syncer_binlog_file{node="master"}`
+- info: 需要配置 `position of binlog position` 一起看. `syncer_binlog_file{node="master"}` 表示上游 MySQL 当前 binlog position 的文件编号, and `syncer_binlog_file{node="syncer"}` 表示上游 Syncer 已经同步到的 binlog 位置的文件编号
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
- info: 需要配置 `position of binlog position` 一起看。`syncer_binlog_file{node="master"}` 表示上游 MySQL 当前 binlog position 的文件编号，`syncer_binlog_file{node="syncer"}` 表示上游 Syncer 已经同步到的 binlog 位置的文件编号
```

File: tools/syncer.md [hunk 15]
```diff
@@ -480,35 +493,51 @@ Syncer 对外提供 metric 接口，需要 Prometheus 主动获取数据。配
...
+
+
+#### title: execution jobs
+
+- metrics: `sum(rate(syncer_add_jobs_total[1m])) by (queueNo)`
+- info: Syncer 把 binlog 转换成 SQLs 后，将 SQLs 以 jobs 的方式加到执行队列中，这个 metrics 表示已经加入执行队列的 jobs 总数
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
- info: Syncer 把 binlog 转换成 SQL 语句后，将 SQL 语句以 jobs 的方式加到执行队列中，这个 metrics 表示已经加入执行队列的 jobs 总数
```

**Comment by GregoryIan** [comment]
@lilin90 PTAL

File: tools/syncer.md [hunk 16]
```diff
@@ -90,49 +98,57 @@ Syncer 的配置文件 `config.toml`：
...
-## 将 127.0.0.1 修改为相应主机 IP 地址
-status-addr = "127.0.0.1:10086"
+## 跳过 DDL 语句，格式为 **前缀完全匹配**，如：`DROP TABLE ABC` 至少需要填入`DROP TABLE`
+# skip-ddls = ["ALTER USER", "CREATE USER"]
 
-# 注意: skip-sqls 已经废弃, 请使用 skip-ddls.
-# skip-ddls 可以跳过与 TiDB 不兼容的 DDL 语句，支持正则语法。
-# skip-ddls = ["^CREATE\\s+USER"]
+## 在使用 route-rules 功能后，
+## replicate-do-db & replicate-ignore-db 匹配合表之后(target-schema & target-table )数值
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
## replicate-do-db & replicate-ignore-db 匹配合表之后 (target-schema & target-table) 数值
```

File: op-guide/migration.md [hunk 17]
```diff
@@ -153,28 +153,51 @@ binlog-gtid = "2bfabd22-fff7-11e6-97f7-f02fa73bcb01:1-23,61ccbb5d-c82d-11e6-ac2e
...
+## 如果设置为 true，Syncer 遇到 DDL 语句时就会停止退出
+stop-on-ddl = false
 
-## 跳过 DDL 或者其他语句，格式为 **前缀完全匹配**，如: `DROP TABLE ABC`,则至少需要填入`DROP TABLE`.
-# skip-sqls = ["ALTER USER", "CREATE USER"]
+## 跳过 DDL 语句，格式为 **前缀完全匹配**，如：`DROP TABLE ABC` 至少需要填入`DROP TABLE`
+# skip-ddls = ["ALTER USER", "CREATE USER"]
 
 ## 在使用 route-rules 功能后，
 ## replicate-do-db & replicate-ignore-db 匹配合表之后(target-schema & target-table )数值
```
Comment (lilin90) [inline op-guide/migration.md]:
```suggestion
## replicate-do-db & replicate-ignore-db 匹配合表之后 (target-schema & target-table) 数值
```

File: tools/syncer.md [hunk 18]
```diff
@@ -90,49 +98,57 @@ Syncer 的配置文件 `config.toml`：
...
+## pprof 调试地址，Prometheus 也可以通过该地址拉取 Syncer metrics
+status-addr = ":8271"
+
+## 如果设置为 true，Syncer 遇到 DDL 语句时就会停止退出
+stop-on-ddl = false
 
-## pprof 调试地址, Prometheus 也可以通过该地址拉取 syncer metrics
-## 将 127.0.0.1 修改为相应主机 IP 地址
-status-addr = "127.0.0.1:10086"
+## 跳过 DDL 语句，格式为 **前缀完全匹配**，如：`DROP TABLE ABC` 至少需要填入`DROP TABLE`
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
## 跳过 DDL 语句，格式为 **前缀完全匹配**，如：`DROP TABLE ABC` 至少需要填入 `DROP TABLE`
```

File: op-guide/migration.md [hunk 19]
```diff
@@ -153,28 +153,51 @@ binlog-gtid = "2bfabd22-fff7-11e6-97f7-f02fa73bcb01:1-23,61ccbb5d-c82d-11e6-ac2e
...
 
-## pprof 调试地址, Prometheus 也可以通过该地址拉取 syncer metrics
-## 将 127.0.0.1 修改为相应主机 IP 地址
-status-addr = "127.0.0.1:10086"
+## 如果设置为 true，Syncer 遇到 DDL 语句时就会停止退出
+stop-on-ddl = false
 
-## 跳过 DDL 或者其他语句，格式为 **前缀完全匹配**，如: `DROP TABLE ABC`,则至少需要填入`DROP TABLE`.
-# skip-sqls = ["ALTER USER", "CREATE USER"]
+## 跳过 DDL 语句，格式为 **前缀完全匹配**，如：`DROP TABLE ABC` 至少需要填入`DROP TABLE`
```
Comment (lilin90) [inline op-guide/migration.md]:
```suggestion
## 跳过 DDL 语句，格式为 **前缀完全匹配**，如：`DROP TABLE ABC` 至少需要填入 `DROP TABLE`
```

File: tools/syncer.md [hunk 20]
```diff
@@ -338,7 +351,12 @@ target-table = "order_2017"
 
 ### Syncer 同步前检查
 
-1. 源库 server-id 检查
+1. MySQL 版本 `select @@version;` Syncer 目前只支持以下版本
+
+    1. MySQL:  5.5 < version < 5.8
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
    - 5.5 < MySQL 版本 < 5.8
```

File: tools/syncer.md [hunk 21]
```diff
@@ -338,7 +351,12 @@ target-table = "order_2017"
...
+    2. MariaDB:  version >= 10.1.2 （更早版本的 binlog 部分字段类型格式与 MySQL 不一致）
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
    - MariaDB 版本 >= 10.1.2（更早版本的 binlog 部分字段类型格式与 MySQL 不一致）
```

File: tools/syncer.md [hunk 22]
```diff
@@ -354,88 +372,109 @@ target-table = "order_2017"
...
+    1. 全量导出的 mydumper 需要的用户权限
 
-6. 检查上下游同步用户权限
+        - mydumper 导出数据至少拥有以下权限：`select, reload`
+        - mydumper 操作对象为 RDS 时，可以添加 `--no-locks` 参数，避免申请 `reload` 权限
 
-    - 需要上游 MySQL 同步账号至少赋予以下权限：
+    2. 增量同步 Syncer 需要的上游 MySQL/MariaDB 用户权限
+
+        - 需要上游 MySQL 同步账号至少赋予以下权限：
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
        需要上游 MySQL 同步账号至少赋予以下权限：
```

File: tools/syncer.md [hunk 23]
```diff
@@ -354,88 +372,109 @@ target-table = "order_2017"
...
+        | ALTER | Tables |
+        | INDEX | Tables |
+
+        在下面为你同步的数据库或者表，执行下面的 GRANT 语句
         ```sql
-        show binlog events in 'mysql-bin.000023' from 136676560 limit 10;
+        GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,ALTER,INDEX  ON db.table TO 'your_user'@'your_wildcard_of_host';
         ```
 
+5. 检查 sql mode, 确认上下游的 sql mode 一致，如果不一致会出现数据同步的错误
```
Comment (lilin90) [inline tools/syncer.md]:
```suggestion
5. 检查 SQL mode，确认上下游的 SQL mode 一致，如果不一致会出现数据同步的错误
```

File: tools/syncer.md [hunk 24]
```diff
@@ -354,88 +372,109 @@ target-table = "order_2017"
...
+        | SELECT | Tables |
+        | INSERT | Tables |
+        | UPDATE | Tables |
+        | DELETE | Tables |
+        | CREATE | Databases,tables |
+        | DROP | Databases, tables |
+        | ALTER | Tables |
+        | INDEX | Tables |
+
+        在下面为你同步的数据库或者表，执行下面的 GRANT 语句
```
Comment (lilin90) [inline tools/syncer.md]: Add a blank line after this line.

Comment (lilin90) [inline tools/syncer.md]:
```suggestion
        为所同步的数据库或者表，执行下面的 GRANT 语句：
```

**Comment by GregoryIan** [comment]
@lilin90 PTAL again

File: tools/syncer.md [hunk 25]
```diff
@@ -353,89 +371,121 @@ target-table = "order_2017"
...
-    ```
+        使用如下命令确认是否开启了 binlog：
 
-3. 检查 MySQL binlog 格式是否为 ROW
+        ```sql
+        mysql> show global variables like 'log_bin';
+        +--------------------+---------+
+        | Variable_name | Value  |
+        +--------------------+---------+
+        | log_bin             | ON      |
```
Comment (amyangfei) [inline tools/syncer.md]: some SQL execution result is not line alignment

Comment (GregoryIan) [reply tools/syncer.md]: fixed

**Comment by amyangfei** [comment]
LGTM

**Comment by csuzhangxc** [comment]
LGTM

**Pull request merged**